sdk/synapse/synapse-artifacts/review/synapse-artifacts.api.md

```diff
@@ -0,0 +1,5227 @@
+    name: string;
+    type: "Container" | "Execution" | "Copy" | "HDInsightHive" | "HDInsightPig" | "HDInsightMapReduce" | "HDInsightStreaming" | "HDInsightSpark" | "ExecuteSSISPackage" | "Custom" | "SqlServerStoredProcedure" | "ExecutePipeline" | "Delete" | "AzureDataExplorerCommand" | "Lookup" | "WebActivity" | "GetMetadata" | "IfCondition" | "Switch" | "ForEach" | "AzureMLBatchExecution" | "AzureMLUpdateResource" | "AzureMLExecutePipeline" | "DataLakeAnalyticsU-SQL" | "Wait" | "Until" | "Validation" | "Filter" | "DatabricksNotebook" | "DatabricksSparkJar" | "DatabricksSparkPython" | "SetVariable" | "AppendVariable" | "AzureFunctionActivity" | "WebHook" | "ExecuteDataFlow" | "SynapseNotebook" | "SparkJob" | "SqlPoolStoredProcedure";
+    userProperties?: UserProperty[];
+}
+
+// @public
+export interface ActivityDependency {
+    [property: string]: any;
+    activity: string;
+    dependencyConditions: DependencyCondition[];
+}
+
+// @public
+export interface ActivityPolicy {
+    [property: string]: any;
+    retry?: any;
+    retryIntervalInSeconds?: number;
+    secureInput?: boolean;
+    secureOutput?: boolean;
+    timeout?: any;
+}
+
+// @public
+export interface ActivityRun {
+    [property: string]: any;
+    readonly activityName?: string;
+    readonly activityRunEnd?: Date;
+    readonly activityRunId?: string;
+    readonly activityRunStart?: Date;
+    readonly activityType?: string;
+    readonly durationInMs?: number;
+    readonly error?: any;
+    readonly input?: any;
+    readonly linkedServiceName?: string;
+    readonly output?: any;
+    readonly pipelineName?: string;
+    readonly pipelineRunId?: string;
+    readonly status?: string;
+}
+
+// @public
+export interface ActivityRunsQueryResponse {
+    continuationToken?: string;
+    value: ActivityRun[];
+}
+
+// @public (undocumented)
+export type ActivityUnion = ControlActivity | ExecutionActivityUnion | ExecutePipelineActivity | IfConditionActivity | SwitchActivity | ForEachActivity | WaitActivity | UntilActivity | ValidationActivity | FilterActivity | SetVariableActivity | AppendVariableActivity | WebHookActivity | SynapseNotebookActivity | SynapseSparkJobDefinitionActivity | SqlPoolStoredProcedureActivity;
+
+// @public
+export type AddDataFlowResponse = AddDataFlowToDebugSessionResponse & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: AddDataFlowToDebugSessionResponse;
+    };
+};
+
+// @public
+export interface AddDataFlowToDebugSessionResponse {
+    jobVersion?: string;
+}
+
+// @public
+export type AmazonMWSLinkedService = LinkedService & {
+    endpoint: any;
+    marketplaceID: any;
+    sellerID: any;
+    mwsAuthToken?: SecretBaseUnion;
+    accessKeyId: any;
+    secretKey?: SecretBaseUnion;
+    useEncryptedEndpoints?: any;
+    useHostVerification?: any;
+    usePeerVerification?: any;
+    encryptedCredential?: any;
+};
+
+// @public
+export type AmazonMWSObjectDataset = Dataset & {
```

Comment:
I'm curious, why is this a using type math instead of doing something like

`export interface AmazonMWSObjectDataset extends Dataset {`